ppl/src/test/java/org/opensearch/sql/ppl/parser/AstBuilderTest.java

```diff
@@ -582,32 +582,88 @@ public void testParseCommand() {
 
   @Test
   public void testKmeansCommand() {
-    assertEqual("source=t | kmeans 3",
-            new Kmeans(relation("t"),exprList(argument("k", intLiteral(3)))));
+    assertEqual("source=t | kmeans centroids=3 iterations=2 distance_type='l1'",
+            new Kmeans(relation("t"), new HashMap<String, Literal>() {{
```

Comment:
I found this subclassing HashMap method is used in UT and elsewhere like MLCommonsOperatorActions. Any reason it's preferred over ImmutableMap?

Reply:
`ImmutableMap` is used in `MLCommonsOperatorActions`. See details [here](https://github.com/opensearch-project/sql/blob/main/opensearch/src/main/java/org/opensearch/sql/opensearch/planner/physical/MLCommonsOperatorActions.java#L9).
Choosing to use `Map` over `ImmutableMap` in other places like `AstBuilder.java` to store command parameters because I found `List` is used for commands instead of `ImmutableList`.

Reply:
Sorry, I meant `new HashMap(){{...}}` to generate an anonymous class may not be necessary.

Reply:
I see, will change `new HashMap` usage in here - https://github.com/opensearch-project/sql/blob/main/opensearch/src/main/java/org/opensearch/sql/opensearch/planner/physical/MLCommonsOperatorActions.java#L53